src/libwalletqt/WalletManager.cpp

```diff
@@ -104,6 +104,20 @@ Wallet *WalletManager::createWalletFromKeys(const QString &path, const QString &
     return m_currentWallet;
 }
 
+Wallet *WalletManager::createWalletFromDevice(const QString &path, const QString &password, NetworkType::Type nettype,
+                                              const QString &deviceName, uint64_t restoreHeight, const QString &subaddressLookahead)
+{
+    QMutexLocker locker(&m_mutex);
+    if (m_currentWallet) {
+        qDebug() << "Closing open m_currentWallet" << m_currentWallet;
+        delete m_currentWallet;
+        m_currentWallet = NULL;
+    }
+    Monero::Wallet * w = m_pimpl->createWalletFromDevice(path.toStdString(), password.toStdString(), static_cast<Monero::NetworkType>(nettype),
+                                                         deviceName.toStdString(), restoreHeight, subaddressLookahead.toStdString());
+    m_currentWallet = new Wallet(w);
```

Comment:
if (w) {
  m_currentWallet = new Wallet(w);
}

Reply:
I'm not sure about your suggestion; this coding style was copied from the existing codebase. If error handling is to be done right, it has to be done thoroughly and simultaneously with all other similar parts.

Reply:
Yep, could be done with separate commit fixing all the similar places. Just thought it would be nice to fix it right here, once you touched this peace of code.

Reply:
Please feel free to make a PR and propose changes about this.